### PR TITLE
Implement C ABI for using lib in other langs

### PR DESCRIPTION
**Pull request by MaximFischuk**

### CODE REVIEW

**Review by StanislavBreadless — APPROVED** [review]
LGTM, waiting for review from @ly0va since he has first-hand experience with musigs

**Review by ly0va — APPROVED** [review]

File: schnorr_musig.h [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+                                                   struct Commitment *commitment);
+
+enum MusigRes schnorr_musig_sign(struct MusigSigner *signer,
+                                 const uint8_t *private_key,
+                                 size_t private_key_len,
+                                 const uint8_t *message,
+                                 size_t message_len,
+                                 struct Signature *signature);
+
+enum MusigRes schnorr_musig_signature_shares(struct MusigSigner *signer,
```
Comment (ly0va) [inline schnorr_musig.h]: This was originally named `receive_signature_shares` though, not sure if a longer name is worth it

Comment (MaximFischuk) [reply schnorr_musig.h]: Oh, this is my fault. I think the better way to leave names the equals to rs.

**Comment by ly0va** [comment]
Who should generally merge these PRs? @popzxc?

**Comment by popzxc** [comment]
> Who should generally merge these PRs? @popzxc?

Either me or @StanislavBreadless

**Pull request merged**